internal/spnego/spnego.go

```diff
@@ -162,7 +163,10 @@ func DecodeNegTokenResp(bs []byte) (*NegTokenResp, error) {
 
 	_, err := asn1.UnmarshalWithParams(bs, &resp, "explicit,tag:1")
 	if err != nil {
-		return nil, err
+		_, err = ber.UnmarshalWithParams(bs, &resp, "explicit,tag:1")
```

Comment:
DER encoding is restricted version of BER encoding, in other words, BER decoder should be able to decode DER encoding.
So, instead of this, simple substitution maybe work:
```
s/asn1\.Unmarshal/ber.Unmarshal/g
``` 
I guess?

Reply:
I was thinking of this as well. Just use `ber.Unmarshal`  instead of `asn1`. 

I saw the comment on testing as well, but since NetApp is a proprietary, licensed tool, I have no way of adding some tests; you would just have to believe me it works.

If you proposed this approach though... Things become more simple.

Reply:
> I saw the comment on testing as well, but since NetApp is a proprietary, licensed tool, I have no way of adding some tests; you would just have to believe me it works.

Hmm, how do you found out that cDot is using BER encoding? I couldn't find any references.
Is it difficult to get a packet capture? I don't think `NegTokenResp` contains sensitive data.

Reply:
We're writing some type of upload solution over the SMB protocol. While testing on local machines everything went well but, when we were given an SMB storage test, we got this `error: superfluous leading zeros in length`. 
After some investigation, we discovered our IT department provided us with an SMB storage location on a NetApp cDOT os.
So we asked our IT department to write to NetApp support, and here is their answer:
_After reviewing the case, we recommend the following action plan: 

According to GoLang the superfluous leading zeros in length error points to OpenStack SSH DER key. 
https://golang.org/src/encoding/asn1/asn1.go?h=superfluous+leading+zeros+in+length 
Please, try to parse the temporary keypair received from OpenStack, if it fails try to use OpenSSL to convert it from BER encoding to DER._

Obviously using OpenSSL was out of the question, so I searched for a go based solution.

Reply:
Thank you for the explanation. But honestly I can't understand connection between SPNEGO and certificates.
Since we're dealing with de/serialization, an unit test with real data is decisive.

Reply:
I understand your reluctance. When I decided to fix this issue, I started debugging to try and discover the line that was throwing the error I have described above, and while trying to connect to NetApp's shared folder, code would always crash at this line https://github.com/hirochachacha/go-smb2/blob/master/internal/spnego/spnego.go#L163 .
This is why, initially, I decided to alter only that specific line.

Reply:
If I think better about this, it might not be a problem of SMB itself, but the login mechanism that's used behind it. While googling about this, I found a lot of complaints about LDAP using BER instead of DER while GoLang's ASN only uses DER (here's an [example](https://groups.google.com/forum/#!topic/golang-nuts/ljEhjkw75-A)). Actually LDAP's [rfc](https://tools.ietf.org/html/rfc4522) is saying communication is encoded using BER.

Here's a link on [LDAPv3 Wire Protocol Reference: The ASN.1 Basic Encoding Rules](https://ldap.com/ldapv3-wire-protocol-reference-asn1-ber/) saying the same thing. And here's a Microsoft [LDAP Representations](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-adts/7cda533e-d7a4-4aec-a517-91d02ff4a1aa) article saying the same thing.

As far as I understood from debugging, that function was crashing during authentification, so it could be LDAP related.

Reply:
I don't know much about LDAP, but we're trying to change SPNEGO code. So let's focus on it.

According to https://tools.ietf.org/html/rfc4178:

> *  DER encoding is required.

Therefore, if some SMB server responds with BER encoding on SPNEGO protocol, that's probably a bug.
That being said, I'm fine with making this library permissive to accept those responses. But still I want some test code.

To be clear, what I'm asking is following data. If it contains sensitive data, you can give me a redacted version.
Will that action be prohibited by the company? Sorry for the repeated request.
![image](https://user-images.githubusercontent.com/898442/89532930-8aaddf00-d82d-11ea-8a79-33776ef72d21.png)

Reply:
Don't think so. And we can redact what may be prohibited. It will take some time to obtain it though :). I'll probably get back to you on Monday or so ...